File: 프로그래머스/lv2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    dic = dict()
    
    for i in range(len(clothes)):
        if clothes[i][1] in dic:
            dic[clothes[i][1]] += 1
        else:
            dic[clothes[i][1]] = 1
    
    for i in dic.values():
        answer *= i + 1
    
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.java
import java.io.*;
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        int n = clothes.length;
        
        HashMap<String, Integer> map = new HashMap<>();
        for(int i=0; i<n; i++) {
            String type = clothes[i][1];
            if(map.containsKey(type)) {
                map.put(type, map.get(type)+1);
            } else {
                map.put(type, 1);
            }
        }
        
        for(Integer value : map.values()) {
            answer *= (value + 1);
        }
        
        return answer - 1;
    }
}